### PR TITLE
feat: add support for handling non standard postgres dates

### PR DESCRIPTION
**Pull request by sighphyre**
Add support for postgres date formats other than "ISO, DMY"

### CODE REVIEW

File: src/lib/create-config.ts [hunk 1]
```diff
@@ -111,6 +117,12 @@ const defaultDbOptions: IDBOption = {
             process.env.DATABASE_POOL_IDLE_TIMEOUT_MS,
             secondsToMilliseconds(30),
         ),
+        ...(parseEnvVarBoolean(
```
Comment (sighphyre) [inline src/lib/create-config.ts:120]: This syntax was new for me, if anyone else doesn't recognise it, this uses the spread operator to optionally assign a property if the boolean condition is truthy. A bit more info here for the curious minded: https://github.com/tc39/proposal-object-rest-spread/issues/45

File: website/docs/deploy/configuring-unleash-v3.md [hunk 2]
```diff
@@ -40,10 +40,11 @@ unleash.start(unleashOptions);
   - _database_ - the database name to be used (`DATABASE_NAME`)
   - _ssl_ - an object describing ssl options, see https://node-postgres.com/features/ssl (`DATABASE_SSL`, as a stringified json object)
   - _version_ - the postgres database version. Used to connect a non-standard database. Defaults to `undefined`, which let the underlying adapter to detect the version automatically. (`DATABASE_VERSION`)
-  - _pool_ - an object describing pool options, see https://knexjs.org/#Installation-pooling. We support the following three fields:
+  - _pool_ - an object describing pool options, see http://knexjs.org/guide/#pool. We support the following four fields:
```
Comment (sighphyre) [inline website/docs/deploy/configuring-unleash-v3.md:43]: This link seems to be dead. The updated link appears to contain what I expect it to contain but more eyes from someone who was familiar with what the old link pointed to would be great!

**Comment by sighphyre** [comment]
This is hard to write an automated test for because it requires configuring a fresh db instance to replicate the issue. Confirmed through local testing and green automated tests

File: website/docs/deploy/configuring-unleash-v3.md (hunk 2 above)
Comment (sighphyre) [reply website/docs/deploy/configuring-unleash-v3.md]: (the diff here is stale, I've updated the links to point to their HTTPS versions)

File: src/lib/create-config.ts (hunk 1 above)
Comment (thomasheartman) [reply src/lib/create-config.ts:120]: Ooh, that's interesting. Looks right, but maybe it's more legible to do:
```js
...(parseEnvVarBoolean(process.env.ALLOW_NON_STANDARD_DB_DATES, false) ? { afterCreate: dateHandlingCallback} : {})
```

Don't have much of a strong preference either way, though 💁🏼

File: website/docs/deploy/configuring-unleash-v3.md [hunk 3]
```diff
@@ -40,10 +40,11 @@ unleash.start(unleashOptions);
...
-  - _pool_ - an object describing pool options, see https://knexjs.org/#Installation-pooling. We support the following three fields:
+  - _pool_ - an object describing pool options, see https://knexjs.org/guide/#pool. We support the following four fields:
     - _min_ - minimum connections in connections pool (defaults to 0) (`DATABASE_POOL_MIN`)
     - _max_ - maximum connections in connections pool (defaults to 4) (`DATABASE_POOL_MAX`)
     - _idleTimeoutMillis_ - time in milliseconds a connection must be idle before being marked as a candidate for eviction (defaults to 30000) (`DATABASE_POOL_IDLE_TIMEOUT_MS`)
+    - \_afterCreate - a callback for for configuring active connections, see https://knexjs.org/guide/#aftercreate. This is incompatible with the `ALLOW_NON_STANDARD_DB_DATES` environment variable, which will override this property to support non standard postgres date formats. If you've set your postgres instance to use a date style other than `ISO, DMY` then you'll need to set the `ALLOW_NON_STANDARD_DB_DATES` environment variable to `true`. Setting the environment variable should be preferred over writing your own callback
```
Comment (thomasheartman) [inline website/docs/deploy/configuring-unleash-v3.md:47]: This is great! 😄 Just a few small suggestions:

```suggestion
    - _afterCreate_ - a callback for for configuring active connections, see https://knexjs.org/guide/#aftercreate. This is incompatible with the `ALLOW_NON_STANDARD_DB_DATES` environment variable, which will override this property to support non-standard Postgres date formats. If you've set your Postgres instance to use a date style other than `ISO, DMY` then you'll need to set the `ALLOW_NON_STANDARD_DB_DATES` environment variable to `true`. Setting the environment variable should be preferred over writing your own callback.
```

I'm assuming this option does not have an env var equivalent (seeing as it's a function)

**Review by thomasheartman — CHANGES_REQUESTED** [review]
Nice one! The docs look mostly good to me (suggestions inline) 🙌🏼  As for the code: that seems surprisingly simple, but that's great! Are the tests you mentioned also included somewhere or are they pre-existing tests?

File: src/lib/create-config.ts (hunk 1 above)
Comment (sighphyre) [reply src/lib/create-config.ts]: Yeah nice, that does look a lot better, I've swapped that out

**Comment by sighphyre** [comment]
@thomasheartman they're existing tests, a whole slew of them go red when the db is configured to use this option without the fix

File: website/docs/deploy/configuring-unleash-v3.md (hunk 3 above)
Comment (sighphyre) [reply website/docs/deploy/configuring-unleash-v3.md]: You are correct sir, there's no equivalent env var for this, the env var mentioned here will override this property to inject the handling function

**Review by thomasheartman — COMMENTED** [review]
Thanks for the fixes! If the existing tests already cover this (as you say they do), I'm happy to let it go through 🥳 You might want someone else to have a look at it, though, but I'll leave that to your discretion.

Just a question: do we allow _any_ non-standard date format now, or just DMY? Could that be opened up for config somehow? Would we even want that?

**Review by thomasheartman — APPROVED** [review]

**Comment by sighphyre** [comment]
@thomasheartman So prior to this patch we support nothing outside of standard format. After this patch, you can set Postgres to use any date format you like and this should handle that. Aside from setting the env var, you shouldn't need any config to make that happen. I don't think there's a need to make this configurable beyond what we have here, it's either default or anything

**Comment by thomasheartman** [comment]
@sighphyre Got it; thanks! That's great, then 😄

**Review by chriswk — COMMENTED** [review]
Since we're setting the date style on the connection, this shouldn't add any surprises for other usages of the same database.

**Pull request merged**